content/advent-2016/contributing-to-the-go-project.md

```diff
@@ -0,0 +1,218 @@
++++
+author = ["Matt Layher"]
+date = "2016-12-08T08:00:00+00:00"
+title = "Contributing to the Go project"
+series = ["Advent 2016"]
++++
+
+For both rookie and veteran Gophers, contributing to the Go project can
+seem overwhelming.  The official
+[Contribution Guidelines](https://golang.org/doc/contribute.html) document
+is rather lengthy; enough so that an
+[issue has been filed](https://github.com/golang/go/issues/17802) to make
+it more clear and concise.
+
+However, after working through the initial Google CLA and Gerrit
+authentication process, it becomes much easier to contribute to the project.
+This post will attempt to demystify the process behind contributing to the
+Go project, in an effort to encourage all Gophers to try to tackle an issue
+or solve a bug upstream for the benefit of others.
+
+Please note: this post attempts to simplify the steps laid out in the
+official Go [Contribution Guidelines](https://golang.org/doc/contribute.html)
+document, but it is entirely possible that it may contain incorrect information
+or become outdated over time.  When in doubt, refer to the Contribution
+Guidelines or consult one of the [many help forums](https://golang.org/help/)
+available for Go.
+
+## Gerrit Registration and Google CLA
+
+The Go project uses a system called [Gerrit](https://www.gerritcodereview.com/)
+for code review.  Gerrit is where CLs (change lists; akin to a GitHub pull
```

Comment:
Rather than keeping the patch history in the tree or lose it completely with a squash, is available in gerrit and the patch is cherry-picked to master.